rest-api/alembic/versions/060dba019a3a_deidentify_work_zip.py

```diff
@@ -0,0 +1,106 @@
+"""deidentify work zip
+
+Revision ID: 060dba019a3a
+Revises: 04564bf1b5e8
+Create Date: 2018-01-04 09:51:00.885405
+
+"""
+from alembic import op
+import sqlalchemy as sa
+import model.utils
+
+
+from participant_enums import PhysicalMeasurementsStatus, QuestionnaireStatus, OrderStatus
+from participant_enums import WithdrawalStatus, SuspensionStatus
+from participant_enums import EnrollmentStatus, Race, SampleStatus, OrganizationType
+from participant_enums import MetricSetType, MetricsKey
+from model.site_enums import SiteStatus
+from model.code import CodeType
+
+# revision identifiers, used by Alembic.
+revision = '060dba019a3a'
+down_revision = '04564bf1b5e8'
+branch_labels = None
+depends_on = None
+
+
+def upgrade(engine_name):
+    globals()["upgrade_%s" % engine_name]()
+
+
+def downgrade(engine_name):
+    globals()["downgrade_%s" % engine_name]()
+
+_QUESTIONNAIRE_RESPONSE_ANSWER_VIEW_SQL = """
+CREATE OR REPLACE VIEW questionnaire_response_answer_view AS
+ SELECT
+   p.participant_id participant_id,
+   YEAR(p.sign_up_time) participant_sign_up_year,
+   p.withdrawal_status participant_withdrawal_status,
+   YEAR(p.withdrawal_time) participant_withdrawal_year,
+   p.suspension_status participant_suspension_status,
+   YEAR(p.suspension_time) participant_suspension_year,
+   hpo.name hpo,
+   (SELECT GROUP_CONCAT(concept_code.value SEPARATOR ',') FROM questionnaire_concept qc
+     INNER JOIN code concept_code ON concept_code.code_id = qc.code_id
+     WHERE qc.questionnaire_id = q.questionnaire_id) module,
+   q.questionnaire_id questionnaire_id,
+   q.created questionnaire_creation_time,
+   qr.questionnaire_response_id questionnaire_response_id,
+   YEAR(qr.created) questionnaire_response_submission_year,
+   qc.value question_code,
+   qc.code_id question_code_id,
+   YEAR(qra.end_time) answer_end_year,
+   ac.value answer_code,
+   ac.code_id answer_code_id,
+   qra.value_boolean answer_boolean,
+   qra.value_decimal answer_decimal,
+   IF(qc.value = 'EmploymentWorkAddress_ZipCode',
+      NULL,
+      qra.value_integer) answer_integer,
+   IF(qra.value_string IS NULL, 0, 1) answer_string_present,
+   YEAR(qra.value_date) answer_date_year,
+   YEAR(qra.value_datetime) answer_datetime_year,
+   IF(qra.value_uri IS NULL, 0, 1) answer_uri_present,
+   IF(qc.value != 'EmploymentWorkAddress_ZipCode', NULL,
```

Comment:
I'm wondering if we can instead of this column just have an answer_integer_present column, which can be used for any integers we don't want to include due to PII concerns.

I don't think we actually need the zip in this table -- we're not doing it for other zip code answers...

Reply:
Or, as discussed over Slack with Regina: just populate answer_integer with the integer version of this.

Adding a new column just for this particular question seems weird.

Reply:
Well, we're adding a new oneoff use column in either case. The zip code column would have applied to any other zip code values we wanted to plumb through deidentified in the future.

That said, I'm happy to go with the simpler option. Done.

Reply:
Annnnd... actually done.